src/ngraph/op/fused/hard_sigmoid.cpp

```diff
@@ -29,41 +29,70 @@ using namespace ngraph;
 
 constexpr NodeTypeInfo op::HardSigmoid::type_info;
 
-op::HardSigmoid::HardSigmoid(const Output<Node>& data, float alpha, float beta)
-    : FusedOp({data})
-    , m_alpha(alpha)
-    , m_beta(beta)
+op::HardSigmoid::HardSigmoid(const Output<Node>& data,
+                             const Output<Node>& alpha,
+                             const Output<Node>& beta)
+    : FusedOp({data, alpha, beta})
 {
     constructor_validate_and_infer_types();
 }
 
+void op::HardSigmoid::pre_validate_and_infer_types()
+{
+    const auto& alpha_pshape = get_input_partial_shape(1);
+    const auto& beta_pshape = get_input_partial_shape(2);
+    NODE_VALIDATION_CHECK(this,
+                          alpha_pshape.is_static() && beta_pshape.is_static(),
```

Comment:
Since FusedOp supports dynamic inputs it's not correct to check that shapes are static. So please handle dynamic shapes also.
BTW if you take a look into validate_and_infer_types method for FusedOp it calls is_dynamic() function and in case of not static shapes it will skip decomposition.

Reply:
I thought about it for a moment and since the specs says that `alpha` and `beta` should be scalars - do we really need to handle dynamic shapes for both of those inputs? I've added this restriction intentionally to reflect the requirements I was given.

Reply:
This scalars can be calculated after constant folding pass for example. So this inputs can be dynamic at some time.

Reply:
Ok, got it. I've relaxed the requirements for alpha and beta.